megaparse/api/app.py

```diff
@@ -84,18 +92,24 @@ async def parse_file(
     )
     try:
         parser = parser_builder.build(parser_config)
-        with tempfile.NamedTemporaryFile(
-            delete=False, suffix=f".{str(file.filename).split('.')[-1]}"
-        ) as temp_file:
-            temp_file.write(file.file.read())
-            megaparse = MegaParse(parser=parser)
-            result = await megaparse.aload(file_path=temp_file.name)
-            return {"message": "File parsed successfully", "result": result}
+
+        megaparse = MegaParse(parser=parser)
+        if not file.filename:
+            raise Exception("No filename provided")
+        _, extension = os.path.splitext(file.filename)
+        file_bytes = await file.read()
+        file_stream = io.BytesIO(file_bytes)
+
+        result = await megaparse.aload(file=file_stream, file_extension=extension)
+        return {"message": "File parsed successfully", "result": result}
     except Exception as e:
```

Comment:
Same here, can we catch a more specific Exception? perhaps @AmineDiro has suggestions

Reply:
I created a ParsingException tell me if that is ok

Reply:
This is at API controller level so I guess we can catch all errors and return a generic exception but we may need to specify other exceptions